services/wallet/metric/metric.go

```diff
@@ -0,0 +1,44 @@
+package metric
+
+import "github.com/prometheus/client_golang/prometheus"
+
+const (
+	cntLinkZP     = "count_link_zebpay"
+	cntLinkHelpZP = "Counts the number of successful and failed ZebPay linkings partitioned by country code"
+	status        = "status"
+	countryCode   = "country_code"
+)
+
+type Metric struct {
+	cntLinkZP *prometheus.CounterVec
+}
+
+func New() *Metric {
+	clzp := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: cntLinkZP,
+		Help: cntLinkHelpZP,
+	},
+		[]string{status, countryCode},
+	)
+	prometheus.MustRegister(clzp)
```

Comment:
Question: How much trouble would it be to make it explicit that this function can fail? I can see the benefit of not having to deal with a returned error. However, the purist in me is happy to accept the complexity of handling it so as to avoid hiding this panic.

Generally, I do not like implicit panics. And libraries that rely on panics are selfish. This `MustRegister` thing [can panic](https://github.com/prometheus/client_golang/blob/80d3f0b5b36c891016370423516e353b1a5f0a55/prometheus/registry.go#L401-L408). Any code that wishes to initialise such metric handler would be susceptible to panicing.

What do you think?